modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/freelist/PagesList.java

```diff
@@ -1995,6 +2002,138 @@ public long metaPageId() {
         return metaPageId;
     }
 
+    /**
+     * Collects the pages that list consists of.
+     *
+     * @param pageIds Initial list of pages ids, that will be added to result collection.
+     * @return Collection of pages that list consists of.
+     */
+    protected Collection<Long> pageIds(long... pageIds) {
```

Comment:
This method seems to bring very little value as it dumps pages that are fine (not corrupted) creating a lot of noise for further investigation of the issue but not making it easier.